_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,539 @@
+
+* If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+* If the throttle is too high, the overall health of the cluster may be impacted.
+
+You can use the `kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)` metric (which with the default `spec.kafka.metrics` settings in your Kafka CR would be named in Prometheus as `kafka_server-fetcherlagmetrics_consumerlag`) to observe how far the followers are lagging behind the leader. You can refer to this [example](https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/examples/metrics/kafka-metrics.yaml) for configuring metrics
+
+The best way to set the value for `--throttle` is to start with a safe value.
+If the lag is growing, or decreasing too slowly to catch up within a reasonable time, then the throttle should be increased.
+To do this, run the command again to increase the throttle, iterating until it looks like the broker will join the ISR within a reasonable time.
+
+The `--verify` mode checks whether all the partitions in have been moved to their intended brokers, as defined by the `reassignment.json` file.
+If the reassignment is complete, `--verify` also removes any replication quotas (`--throttle`) that are in effect.
+Unless removed, throttles will continue to affect the cluster even after the reassignment has finished.
+
+To summarise, a typical reasignment might look like this:
+
+1. Start the reassignment: `kafka-reassign-partitions.sh --execute --throttle 5000000 ...`
+2. Observe a growing lag
+3. Increase the throttle: `kafka-reassign-partitions.sh --execute --throttle 8000000 ...`
+4. Maybe now the lag is falling, but too slowly to expect the reassingment to be complete within a tolerable time.
+5. So increase the throttle agin: `kafka-reassign-partitions.sh --execute --throttle 10000000 ...`
+6. Check for completion: `kafka-reassign-partitions.sh --verify ...`, but it's not complete.
+7. Check for completion: `kafka-reassign-partitions.sh --verify ...`, this time the reassignment is complete, so this invocation will have removed the throttle.
+
+### Setting up the environment
+
+Let's spin up an cluster where we can work through some reassignment examples.
+
+To get the Kafka cluster up and running, we will first have to install the Strimzi Cluster Operator and then deploy the Kafka resource.
+You can refer to the [Stimzi Quickstart Guide](https://strimzi.io/docs/operators/latest/quickstart.html) for installing Strimzi.
+
+You can install the Cluster Operator with any installation method you prefer.
+The Kafka cluster is then deployed with the plain listener enabled on port 9092.
+
+Example Kafka configuration with plain listener.
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+   name: my-cluster
+spec:
+   kafka:
+      version: 3.2.1
```

Comment:
At this point, 3.2.1 is not released. Maybe you can change it to 3.2.0? Or remove the version completely? That will just keep it valid for the future.